### PR TITLE
Plugin http api

### PR DESCRIPTION
**Pull request by zhongwencool**
This PR only solved the API of single-node plugin.
The mechanism of adding synchronization plugins to new nodes and state synchronization will be implemented in future PRs.

### CODE REVIEW

File: apps/emqx_management/src/emqx_mgmt_api_plugin.erl [hunk 1]
```diff
@@ -0,0 +1,328 @@
...
+    {GoodRes, BadNodes} = rpc:multicall(Nodes, Mod, Fun, Args, Timeout),
+    BadNodes =/= [] andalso
+        ?LOG(error, "rpc call failed on ~p ~p:~p/~p", [BadNodes, Mod, Fun, length(Args)]),
+    GoodRes.
+
+cluster_rpc(Mod, Fun, Args) ->
+    case emqx_cluster_rpc:multicall(Mod, Fun, Args, all, 30000) of
+        {ok, _TnxId, Res} -> Res;
+        {retry, TnxId, Res, Node} ->
+            ?SLOG(error, #{msg => "failed to update plugin in cluster", nodes => Node,
```
Comment (zmstone) [inline apps/emqx_management/src/emqx_mgmt_api_plugin.erl:312]: it's `Nodes` not `Node`?
```suggestion
            ?SLOG(error, #{msg => "failed to update plugin in cluster", bad_nodes => Nodes,
```

File: apps/emqx_plugins/src/emqx_plugins_monitor.erl [hunk 2]
```diff
@@ -0,0 +1,83 @@
+%%--------------------------------------------------------------------
+%% Copyright (c) 2021 EMQ Technologies Co., Ltd. All Rights Reserved.
```
Comment (zmstone) [inline apps/emqx_plugins/src/emqx_plugins_monitor.erl:2]: 2022

File: apps/emqx_plugins/src/emqx_plugins_monitor.erl [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+handle_call(Req, _From, State) ->
+    ?SLOG(error, #{msg => "unexpected_call", call => Req}),
+    {reply, ignored, State}.
+
+handle_cast(Msg, State) ->
+    ?SLOG(error, #{msg => "unexpected_cast", cast => Msg}),
+    {noreply, State}.
+
+handle_info({timeout, Ref, check}, State = #{failed := Failed}) ->
+    erlang:cancel_timer(Ref),
```
Comment (zmstone) [inline apps/emqx_plugins/src/emqx_plugins_monitor.erl:44]: this timer is clearly expired.
no need to cancel.?

File: apps/emqx_plugins/src/emqx_plugins_monitor.erl [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+    ?SLOG(error, #{msg => "unexpected_info", info => Info}),
+    {noreply, State}.
+
+terminate(_Reason, _State) ->
+    ok.
+
+code_change(_OldVsn, State, _Extra) ->
+    {ok, State}.
+
+next_check_time() ->
```
Comment (zmstone) [inline apps/emqx_plugins/src/emqx_plugins_monitor.erl:57]:
```suggestion
schedule_next_check() ->
```

**Review by DDDHuang — APPROVED** [review]

**Pull request merged**